content/docs/buildpack-author-guide/caching-strategies.md

```diff
@@ -0,0 +1,65 @@
+* `launch` layers -- the directory will be included in the run image as a single layer,
+
+A fourth type of layer
+
+* `ignored` layers
+
+are available to buildpack authors for use as temporary layers.
+
+In this section we look at caching each layer type.
+
+## Layer Metadata
+
+Buildpacks ensure byte-for-byte reproducibility of layers.  File creation time is normalized to January 1, 1980 to ensure reproducibility.  Byte-for-byte reproducibility means we can re-use previous layers.  However, we want to invalidate previously cached layers if
+
+* the buildpacks API changes,
+* the type of the layer changes.
+
+A layer built using a buildpack at API version `0.7` should be considered invalid if the API version for that buildpack has been updated to `0.8`.  Similarly, if a layer is changed from being a cache-only layer to being a cache and launch layer, then the cache should be considered invalid.
+
+In addition to general cache invalidation conditions our pip buildpack should invalidate a previous layer if an important property changes, such as:
+
+* the major version of the runtime changes eg: NodeJS changes from 16 to 18
+* requested application dependencies have changed eg: a Python application adds a dependecy on the `requests` module
+
+Launch layers are exported to an OCI registry and layer metadata is stored with the launch layer.  The layer metadata is commonly used when deciding if a launch layer should be re-used from cache.
+
+## Strategies
+
+Caching during the production of an application image is necessarily very flexible.  Most buildpacks that wish to contribute a layer to the application image need only to
+
+1. Check that the metadata of the cached layer is current,
+2. Create an empty layer, and
+3. Set `Launch = true`.
+
+This will guarantee that the previously published application image layer in the registry is re-used if the layer metadata matches the requested metadata.  In this most straightforward use-case `Launch` is `true` and both `Build` and `Cache` are set to `false`.  That is to say, the most common case is where `Cache = false`, `Build = false` and `Launch = true`.  It is important to note that the layer is re-used on on the OCI registry.  This avoids expensive rebuilds of the layer and expensive pulls of the layer to the host running the build.
+
+Setting `Build = true` makes a layer available to subsequent buildpacks.  Therefore binaries installed to the `bin` directory on a `Build = true` layer are available to subsequent buildpacks during the build phase.  It is also the case that `lib` directories on a `Build = true` later are added to the `LD_LIBRARY_PATH` during the build phase of subsequent buildpacks.  Environment variables defined in a `Build = true` layer are similarly available.  For any layer where `Launch = true` and `Build = true` we can no-longer re-use a launch layer from the OCI registry; we must make the layer available locally so that subsequent buildpacks can use it.
```

Comment:
> For any layer where `Launch = true` and `Build = true` we can no-longer re-use a launch layer from the OCI registry; we must make the layer available locally so that subsequent buildpacks can use it.

For any layer where `Launch = true` and `Build = true", a launch layer from the OCI registry can no longer be reused. Instead, the layer must be made available locally so that subsequent buildpacks can use it.
(avoding the use of first-person)

Reply:
Thanks